src/Symfony/Component/DependencyInjection/Compiler/ResolveReferencesToAliasesPass.php

```diff
@@ -76,6 +78,21 @@ private function processArguments(array $arguments)
         return $arguments;
     }
 
+    private function processFactory($factory)
+    {
+        if (null === $factory || !is_array($factory) || !$factory[0] instanceof Reference) {
+            return $factory;
+        }
+
+        $defId = $this->getDefinitionId($id = (string) $factory[0]);
+
+        if ($defId !== $id) {
+            $factory[0] = new Reference($defId, $factory[0]->getInvalidBehavior(), $factory[0]->isStrict());
```

Comment:
note that this line will need changes when merging into 2.8 (and then again in 3.0) due to changes to ``isStrict``